src/component/document.ts

```diff
@@ -110,60 +101,70 @@ export const remove = mutation({
   returns: v.boolean(),
   handler: async (ctx, args) => {
     const s2 = await S2Bindings.load();
+    const { success } = await removePointByKey(ctx, args.key, args, s2);
+    return success;
+  },
+});
 
-    const existing = await ctx.db
-      .query("points")
-      .withIndex("key", (q) => q.eq("key", args.key))
-      .first();
-    if (!existing) {
-      return false;
-    }
+async function removePointByKey(
+  ctx: MutationCtx,
+  key: string,
+  opts: {
+    minLevel: number;
+    maxLevel: number;
+    levelMod: number;
+    maxCells: number;
+  },
+  s2: S2Bindings
+): Promise<{ success: boolean; existing?: any }> {
```

Comment:
remove `existing` from the return value -- it's not needed in either caller

change the return value to just be a `boolean`